### ISSUE TITLE
Does the entire path need to be valid in order to use a path attribute on a textPath

### ISSUE THREAD

**Issue by longsonr** [w3c/svgwg#393]
Per https://www.w3.org/TR/SVG2/text.html#TextPathElement

If the ‘path’ attribute contains an error, the ‘href’ attribute must be used. 

I can understand that path="" or path="foo" might be ignored.

However, does this also mean that a path="M 100, 100 foo" is ignored? That seems inconsistent with path error handling where it says that they are rendered until an error occurs (https://www.w3.org/TR/SVG2/paths.html#PathDataErrorHandling)

I propose that path attributes be handled the same way as path rendering i.e. as long as the first element in the path is valid, the path would be used.

**Comment by AmeliaBR** [comment]
I agree with your logic, and I don't think we have any implementations yet, so there is no compat risk in changing it.

Furthermore, we have this instruction in the definition of `path` attribute:

> An empty string indicates that there is no path data for the element. This means that the text within the ‘textPath’ does not render or contribute to the bounding box of the ‘text’ element.  If the attribute is not specified, the path specified with ‘href’ is used instead.

If that is the desired behavior (similar to an empty `d` value on a `<path>` preventing rendering), then I think that should be the invalid value default as well (that is, `path="foo"` should behave the same as `path=""`).  And then it only makes sense to use partial parsing, too. Basically, if a `path` attribute is present, `href` is ignored. Use as much of the path as can be parsed, and then layout as much of the text as fits on that path..

For the summary description, the "If the path contains an error..." sentence could be removed completely.
_______________________

Aside, when someone makes edits, also fix the typo in the `href` definition:

> or the referenced element is not a ‘path’ element) or basic shape

"or basic shape" should be inside the ")"!

**Comment by boggydigital** [comment]
Not blocking updated 2.0 CR publication - assigning 2.1 WD milestone

**Comment by yisibl** [comment]
> For the summary description, the "If the path contains an error..." sentence could be removed completely.

I agree to remove this sentence. Currently, neither Chrome nor Firefox will ignore an incorrect path value.

See also:
- https://github.com/linebender/resvg/issues/417#issuecomment-3852266627
- https://issues.chromium.org/issues/374010056


@karlcow Do you have time to take a look at this?

**Comment by karlcow** [comment]
https://w3c.github.io/svgwg/svg2-draft/text.html#:~:text=attribute%20contains%20an%20error%2C

The sentences is still in there:
> If the ‘[path](https://w3c.github.io/svgwg/svg2-draft/text.html#TextPathElementPathAttribute)’ attribute contains an error, the ‘[href](https://w3c.github.io/svgwg/svg2-draft/text.html#TextPathElementHrefAttribute)’ attribute must be used.

Yes we could remove it.

From https://w3c.github.io/svgwg/svg2-draft/text.html#TextPathElement
1. Click on path https://w3c.github.io/svgwg/svg2-draft/text.html#TextPathElementPathAttribute
2. Click on path data https://w3c.github.io/svgwg/svg2-draft/paths.html#PathDataBNF

At the bottom, it says:

> The EBNF allows the path data string in the [d](https://w3c.github.io/svgwg/svg2-draft/paths.html#DProperty) property to be empty. An empty path data string disables rendering of the path. Rendering is also disabled when the [d](https://w3c.github.io/svgwg/svg2-draft/paths.html#DProperty) property has the value none.
> 
> If path data not matching the grammar is encountered, then the path data is in error (see [Error Handling](https://w3c.github.io/svgwg/svg2-draft/paths.html#PathDataErrorHandling)).

3. Click on error handling https://w3c.github.io/svgwg/svg2-draft/paths.html#PathDataErrorHandling

which addresses the issues about invalid path data.

**Comment by karlcow** [comment]
This seems a separate issue and for the SVG WG.
fyi, there is a [`path()` in CSS](https://drafts.csswg.org/css-shapes/#funcdef-basic-shape-path)

**Comment by karlcow** [comment]
@yisibl Is there already full coverage of the different scenarios for valid and invalid path in SVG in the context of textPath.

If not, that would be the first steps, create WPT testcases to understand, what browsers are currently doing, and see if there is an interop path possible.

**Comment by viralipurbey** [comment]
yes I would love to add them. I believe once we come to a conclusion about all the valid case scenarios, I'll add wpts for the same to cover all the cases.

**Comment by dirkschulze** [comment]
https://www.w3.org/2026/02/19-svg-minutes.html
RESOLUTION: Drop the sentence where the reference should be used on invalid path. Instead use the path attribute and path data up to the first error. Even if the error results in no path at all. path takes precedence over reference.

**Comment by viralipurbey** [comment]
Added WPTs for this resolution as discussed in the meeting. https://github.com/web-platform-tests/wpt/pull/58166

**Comment by dirkschulze** [comment]
RESOLUTION: If the d attribute would result into an empty path (not specified, no path data or due to an error) we fallback to href. If there is at least some valid path data at the beginning, we will continue to use the path data specified with the d attribute up to the error.

https://www.w3.org/2026/03/12-svg-minutes.html
